pyvista/plotting/lights.py

```diff
@@ -0,0 +1,466 @@
+        defined in a coordinate space where the camera is located at
+        (0, 0, 1), looking towards (0, 0, 0) at a distance of 1, with
+        up being (0, 1, 0). Camera lights use the transform matrix to
+        establish this space.
+
+        The property returns class constant values from an enum:
+            - Light.HEADLIGHT == 1
+            - Light.CAMERA_LIGHT == 2
+            - Light.SCENE_LIGHT == 3
+
+        """
+        return LightType(self.GetLightType())
+
+    @light_type.setter
+    def light_type(self, ltype):
+        """Set the light type.
+
+        Either an integer code or a class constant enum value must be used.
+
+        """
+        if not isinstance(ltype, int):
+            # note that LightType is an int subclass
+            raise TypeError('Light type must be an integer subclass,'
+                            f' got {ltype} instead.')
+        self.SetLightType(ltype)
+
+    @property
+    def is_headlight(self):
+        """Return whether the light is a headlight."""
+        return bool(self.LightTypeIsHeadlight())
+
+    @property
+    def is_camera_light(self):
+        """Return whether the light is a camera light."""
+        return bool(self.LightTypeIsCameraLight())
+
+    @property
+    def is_scene_light(self):
+        """Return whether the light is a scene light."""
+        return bool(self.LightTypeIsSceneLight())
+
+    @property
+    def shadow_attenuation(self):
+        """Return the shadow intensity.
+
+        By default a light will be completely blocked when in shadow.
+        By setting this value to less than 1 you can control how much
+        light is attenuated when in shadow.
+
+        """
+        return self.GetShadowAttenuation()
+
+    @shadow_attenuation.setter
+    def shadow_attenuation(self, shadow_intensity):
+        """Set the shadow intensity."""
+        self.SetShadowAttenuation(shadow_intensity)
+
+    #### Everything else ####
+
+    def set_color(self, color):
+        # TODO: do we want to remove this?
+        # TODO: do we want instead to set _all_ lights in pyvista?
+        # TODO: do we want to support a three-arg form?
+        """Set the diffuse and specular color of the light.
+
+        Parameters
+        ----------
+        color : tuple(float)
+            The color that should be set for diffuse and specular.
+
+        """
+        self.SetColor(color)
+
+    def switch_on(self):
+        """Switch on the light."""
+        self.SwitchOn()
+
+    def switch_off(self):
+        """Switch off the light."""
+        self.SwitchOff()
+
+    def switch(self, is_on=None):
+        """Set whether the light should be on. Toggles by default.
+
+        Parameters
+        ----------
+        is_on : bool
+            The state of the light to be set (True is on, False is off).
+            Default is to toggle the state.
+
+        """
+        if is_on is None:
+            # toggle
+            is_on = not self.is_on
+        self.SetSwitch(is_on)
```

Comment:
```suggestion
```
While I like keeping API compatibility with VTK, having both the `on` setter and `switch` seems redundant.  Additionally, were we to keep this, something like `toggle` makes more sense.

For simplicity, lets just remove this and keep just `on`